Add UXPL to edx-platform: Package Usage & Web Certs Syntax Clean Up

talbs opened the pull request:
This work does a few things:

* Updates the npm-based UXPL package reference to 0.8.12
* Removes static/old versions of UXPL code from the platform
* Syncs up any syntax or configuration changes needed for Open Web Certs to leverage the UXPL npm package.

- - -

#### Reviewers

* [ ] Overall - @andy-armstrong

andy-armstrong commented on lms/static/certificates/sass/_config.scss:
```diff
@@ -0,0 +1,37 @@
+// ------------------------------
+// #VARIABLES
+// ------------------------------
+$em-base:       16; // deliberately sets bourbon-based em-base (http://bourbon.io/docs/#em-base)
```
@talbs Can you give myself and @dsjen more context here. Why are you having to set the em-base here?

talbs replied:
Sure thing!

``$em-base`` is a [configuration variable in the UXPL](https://github.com/edx/ux-pattern-library/blob/master/pattern-library%2Fsass%2Fglobal%2F_settings.scss#L27) (that this app is overriding/customizing it). This configuration is used in any of our (Bourbon-based) rem-to-px Sass utilities (see our [font-size function](https://github.com/edx/ux-pattern-library/blob/master/pattern-library%2Fsass%2Fglobal%2F_functions.scss#L20)).

Since the value is the same as the default value used in the UXPL, this line isn't needed. But if an app ever wanted to set its own different base (other than the current, ~16px), this is how to do it per app Sass/CSS. I'll remove this line since its redundant for this case.

andy-armstrong commented on lms/static/certificates/sass/_ltr.scss:
```diff
@@ -0,0 +1,23 @@
+// ------------------------------
```
I thought that this partial could be imported directly from the UXPL. Why do we need our own copy? I was hoping that references to bi-app would be kept inside the UXPL.

talbs replied:
I hear you, but sadly the ``_ltr.scss`` and ``_rtl.scss`` partials need to be saved and pulled in at the application Sass/CSS level that's extending the UXPL Sass for a couple of reasons:

1. We don't know where an individual app will store its npm-installed libraries, which includes any libs pulled in as UXPL dependencies. So a path made in the UXPL assets wouldn't be able to account for that until we can pass variables into an ``@import`` path (see https://github.com/sass/sass/issues/279).
2. The ``$layout-direction: ltr;``, ``$grid-direction-default: ltr;``, and ``$grid-direction-reversed: ltr;`` variables need to be set specifically for an app's main ``*-ltr.scss`` and ``*--rtl.scss`` compile files. I had originally had this logic in the UXPL using the ``@if`` + ``@import`` method in the UXPL that we removed recently (https://github.com/edx/ux-pattern-library/pull/263).

andy-armstrong commented on lms/static/certificates/sass/_rtl.scss:
```diff
@@ -0,0 +1,23 @@
+// ------------------------------
```
Ditto for my question on _ltr.scss

talbs replied:
See my [note above](https://github.com/edx/edx-platform/pull/11307#discussion_r51127756)

andy-armstrong commented on lms/static/certificates/sass/main-rtl.scss:
```diff
@@ -5,18 +5,12 @@
+// #CONFIG - layout direction
 // ------------------------------
-@import '../../../../static/pattern-library/sass/utilities/variables-rtl';
+@import 'ltr';                        // LTR-specifc settings and utilities
```
This should be 'rtl'

talbs replied:
#typotalbs - will fix up

andy-armstrong commented:
@talbs This looks excellent. I have a couple of questions, and then we need to work out if any of the test failures are real, or if they are all related to the SASS compilation.

andy-armstrong commented:
@talbs I rebased to pick up changes to my branch.

talbs commented:
@andy-armstrong, I addressed all of your feedback and squashed changes down into my original commit. Shout if you need anything else and let me know how you want to tackle next steps with using this work. :)

andy-armstrong commented:
:+1: sweet work @talbs. How will we manage without you? :-( I'm going to merge this down to my branch now.
